src/arch/arm/head.S

```diff
@@ -4,53 +4,123 @@
  * @date 28.03.10
  * @author Anton Kozlov
  */
+
+#include <framework/mod/options.h>
+
+#define MMU_BOOT OPTION_GET(BOOLEAN,mmu_boot)
+
 .weak hardware_init_hook
 .weak software_init_hook
 .weak kernel_start
 
+#if MMU_BOOT
+#include <asm/cp15.h>
+/* Put tables to text section to avoid setting it to zero during clearing bss */
+.text
+.align 14
+boot_tables: .space 1024 * 16
+
+.text
+boot_mmu_setup:
+	/* Setup MMU base */
+	ldr r0, =boot_tables
+	mcr p15, 0, r0, c2, c0, 0
+
+	/* Map whole address space 1-1 */
+	mov r1, #0
+	mov r2, #0x1000
+l1:
+	/* Build supersection desc in R3 */
+	orr r3, r1, # 0x6
+	orr r3, r3, # 0xC00
+	#orr r3, r3, # (1 << 18)
```

Comment:
It's required or the same as line above?

Reply:
Removed.